modules/utxo-lib/src/bitgo/UtxoPsbt.ts

```diff
@@ -241,6 +253,24 @@ export class UtxoPsbt<Tx extends UtxoTransaction<bigint>> extends Psbt {
     return results.every((res) => res === true);
   }
 
+  /**
+   * @param publicKeys
+   * @return array of boolean values. True when corresponding index in `publicKeys` has signed the transaction.
+   */
+  getSignatureValidationArray(inputIndex: number, publicKeys: Buffer[]): boolean[] {
+    const input = checkForInput(this.data.inputs, inputIndex);
+    const isP2tr = input.tapScriptSig?.length;
+    return publicKeys.map((publicKey) => {
+      try {
+        return isP2tr
+          ? this.validateTaprootSignaturesOfInput(inputIndex, publicKey)
+          : this.validateSignaturesOfInput(inputIndex, (p, m, s) => eccLib.verify(m, p, s), publicKey);
+      } catch (err) {
+        return false;
+      }
```

Comment:
Not a fan of this overly broad `catch` here. There are a number of errors that can stem from implementation errors unrelated to sig verification that we want to know about. Without the `throw` in `this.validateTaprootSignaturesOfInput` I believe we can do without the `catch` here.

Reply:
this comment remains valid - to be compatible with https://github.com/BitGo/bitcoinjs-lib/blob/master/ts_src/psbt.ts#L435 we should remove the `catch`

Reply:
Errors for unsigned input or input with no matching signature for given public key are caught and returned as false. All other errors, propagating the error.